pipeline_dp/pld_budget_accounting.py

```diff
@@ -0,0 +1,114 @@
+"""Privacy Loss Distribution (PLD) budget accounting for DP Pipelines."""
+
+from dataclasses import dataclass, field
+from pipeline_dp.aggregate_params import NoiseKind
+from dp_accounting import privacy_loss_distribution
+
+
+@dataclass
+class PLDBudget:
+    _noise_parameter: float = None  # i am not sure how to use this, as noise is calibrated globally?
+
+    @property
+    def noise_parameter(self):
+        if self._noise_parameter is None:
+            raise AssertionError("Noise parameter is not yet defined.")
+        return self._noise_parameter
+
+
+@dataclass
+class RequestedPLDBudget:
+    budget: PLDBudget
+
+
+@dataclass
+class PLDBudgetAccountant:
+    noise: float = 1  # this is where i am storing noise, does this make sense?
+    max_noise: float = 50
+    sensitivity: float = 1
+    total_epsilon: float = None
+    total_delta: float = None
+    _requested_laplace_budgets: list = field(default_factory=list)
+    _requested_gaussian_budgets: list = field(default_factory=list)
+
+    def __post_init__(self):
+        if self.total_epsilon is None or self.total_delta is None:
+            raise ValueError(f"Must pass total_epsilon and total_delta")
+        if self.total_epsilon is not None and self.total_epsilon <= 0:
+            raise ValueError(
+                f"Epsilon must be positive, not {self.total_epsilon}.")
+        if self.total_delta is not None and self.total_delta < 0:
+            raise ValueError(
+                f"Delta must be non-negative, not {self.total_delta}.")
+        if self.noise > self.max_noise:
+            raise ValueError(
+                f"Noise = {self.noise} is larger than the maximum = {self.max_noise}."
+            )
+
+    def request_budget(self, noise_kind: NoiseKind) -> PLDBudget:
```

Comment:
Sensitivity is the part of a mechanism specification, so I'd suggest to replace
   NoiseKind ->  
```
class MechanismSpec:  # @dataclass or maybe namedtuple
   NoiseKind
   sensitivity
   weight
``` 
weight is similar to [weight]( # similar to [weight](https://github.com/OpenMined/PipelineDP/blob/65d00ee6c906a36b8a173ddbb83d7e3bf71905b4/pipeline_dp/budget_accounting.py#L48)) in naive accountant

Reply:
I believe it is using weight correctly now:

`pld.PrivacyLossDistribution.from_laplace_mechanism(
                    noise / mechanism.weight, sensitivity=mechanism.sensitivity)`